client/my-sites/theme/main.jsx

```diff
@@ -790,8 +790,11 @@ class ThemeSheet extends Component {
 		if ( ! isLoggedIn ) {
 			plansUrl = localizeUrl( 'https://wordpress.com/pricing' );
 		} else if ( siteSlug ) {
+			const redirectTo = `/theme/${ themeId }${ section ? '/' + section : '' }/${ siteId }`;
 			const plan = isExternallyManagedTheme || isBundledSoftwareSet ? PLAN_BUSINESS : PLAN_PREMIUM;
-			plansUrl = plansUrl + `/${ siteSlug }/?plan=${ plan }`;
+			plansUrl =
+				plansUrl +
+				`/${ siteSlug }/?plan=${ plan }&redirect_to=${ encodeURIComponent( redirectTo ) }`;
```

Comment:
Nitpick: `encodeURIComponent` replaces `/` with `%2F` (e.g. `?redirect_to=%2Ftheme%2Fastra%2Fexample.wordpress.com`), but slashes don't need to be escaped (`?redirect_to=/theme/astra/example.wordpress.com` works too). Since `redirectTo` is defined a few lines above (i.e. it is not given as param), we can guarantee it doesn't include any character that needs to be escaped and simplify this a little bit.
```suggestion
				`/${ siteSlug }/?plan=${ plan }&redirect_to=${ redirectTo }`;
```